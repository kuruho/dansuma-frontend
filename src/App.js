import React from "react";
import {
  Text,
  View,
  TouchableHighlight,
  ScrollView,
  StatusBar,
  ActivityIndicator,
  Dimensions,
  Platform,
  Image
} from "react-native";

// Components
import Display from "./Display";
import SongButton from './SongButton'
// assets
import { color } from "react-native-material-design-styles";
import logo from "./y18.gif";
import styles from "./AppStyles";

// utils
import { domainUrl } from "./utils";
import moment from "moment";

// ComponentsmarginTop:
const Card = ({ children }) => <View style={styles.card}>{children}</View>;

const cursorStyle = Platform.OS === "web" ? { cursor: "pointer" } : {};

const Overlay = ({ children, visible }) =>
  visible
    ? <View
        style={{
          position: "absolute",
          marginTop: Platform.OS === "ios" ? 48 : 28,
          zIndex: 1
        }}
      >
        <Text
          style={{
            fontSize: 28,
            marginLeft: Dimensions.get("window").width * 0.9,
            backgroundColor: "rgba(52,52,52, 0.0)",
            color: "white",
            zIndex: 3
          }}
        >
          ⏏
        </Text>
        <View
          style={{
            padding: 10,
            marginTop: -17,
            width: Dimensions.get("window").width,
            height: 400,
            backgroundColor: "rgba(255,255,255,80)"
          }}
        >
          {children}
        </View>
      </View>
    : <View />;

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.scrollToTop = this.scrollToTop.bind(this);
    // this.voteSong = this.voteSong.bind(this);
    // this.voteSong = props.voteSong
    // console.log(props.voteSong)
  }

  scrollToTop() {
    this.refs._scrollView.scrollTo({ x: 0, y: 0, animated: true });
  }

  render() {
    const {
      items,
      errors,
      loading,
      filter,
      overlayVisible,
      onLoadMore,
      onLoadItems,
      onOpenUrl,
      onToggleOverlay,
      voteSong,
      votedColor,
      votedSong,
      songs
    } = this.props;


    const currentSong = {
      title: "Il cielo in una stanza",
      author: "Albano",
      color: "yellow",
      id: 123
    };

    return (
      <Card>
        {Platform.OS === "android"
          ? <StatusBar backgroundColor={"#d25500"} />
          : <View />}
        <View style={styles.body}>

          <Overlay visible={overlayVisible}>
            <Text style={{ fontSize: 18 }}>
              <Text style={{ fontWeight: "bold" }}>Gioplayer</Text>
              {" "}
              is a simple Hacker News reader for the Web and a React Native app (Android / iOS).
            </Text>
            <Text style={{ fontSize: 18, marginTop: 20 }}>
              Made with ❤ by
              <Text
                onPress={() => {
                  onOpenUrl("https://grigio.org");
                }}
              >
                <Text style={[styles.aboutLink, cursorStyle]}>
                  {" "}Luigi Maselli
                </Text>
              </Text>
              , source code:
              <Text
                onPress={() => {
                  onOpenUrl("https://github.com/grigio/HAgnostic-News");
                }}
              >
                <Text style={[styles.aboutLink, cursorStyle]}>
                  {" "}github.com/grigio/HAgnostic-News
                </Text>
              </Text>
            </Text>
          </Overlay>

          <View
            style={[
              styles.column,
              styles.header,
              Platform.OS === "ios" ? { height: 75, paddingTop: 20 } : {}
            ]}
          >
            <View style={[styles.row, { height: 50 }]}>
              <View style={styles.row}>
                <Image source={logo} style={{ width: 20 }} />
                <Text style={[{ fontWeight: "bold", paddingLeft: 4 }]}>
                  Gioplayer
                </Text>
                <Text style={[{ fontSize: 12, paddingLeft: 4 }]}>
                  {" "}{Platform.OS}
                </Text>
              </View>
              <TouchableHighlight
                style={[
                  styles.button,
                  filter === "Top" ? styles.buttonOrange : null
                ]}
                underlayColor={color.paperOrange200.color}
                onPress={() => {
                  onLoadItems("Top");
                  this.scrollToTop();
                }}
              >
                <View style={styles.row}>
                  <Text
                    style={{
                      color: "white",
                      fontWeight: "bold",
                      paddingRight: 5
                    }}
                  >
                    Top
                  </Text>
                  {filter === "Top" && loading ? <ActivityIndicator /> : null}
                </View>
              </TouchableHighlight>
              <TouchableHighlight
                style={[
                  styles.button,
                  filter === "Latest" ? styles.buttonOrange : null
                ]}
                underlayColor={color.paperOrange200.color}
                onPress={() => {
                  onLoadItems("Latest");
                  this.scrollToTop();
                }}
              >
                <View style={styles.row}>
                  <Text
                    style={{
                      color: "white",
                      fontWeight: "bold",
                      paddingRight: 5
                    }}
                  >
                    Latest
                  </Text>
                  {filter === "Latest" && loading
                    ? <ActivityIndicator />
                    : null}
                </View>
              </TouchableHighlight>
              <TouchableHighlight
                underlayColor={color.paperOrange200.color}
                onPress={() => {
                  onToggleOverlay();
                }}
              >
                <Text
                  style={{ color: "white", fontWeight: "bold", padding: 10 }}
                >
                  ?
                </Text>
              </TouchableHighlight>
            </View>
            {Object.keys(errors).length > 0
              ? <View style={[styles.row, { flex: 1, backgroundColor: "red" }]}>
                  {Object.keys(errors).map((error, i) => (
                    <Text key={i}>. {errors[error].message}</Text>
                  ))}
                </View>
              : null}

          </View>

          <View style={styles.scrollViewContainer}>
            {/* Display */}
            <Display currentSong={currentSong} votedColor={votedColor}/>
            {/*List songbuttons*/}
            <View style={{padding: 2, backgroundColor:'rgb(253, 255, 174)'}}>
              Vote a song below to suggest the next one!
            </View>
            <View style={{ backgroundColor: "#dadada", flex:1, justifyContent:'space-around' }}>
              {/*Songbutton*/}
              {songs.map(song => (
                <SongButton song={song} voteSong={voteSong} selected={song.id === votedSong }/>
              ))}
            </View>
          </View>

        </View>
      </Card>
    );
  }
}

                {/*<TouchableHighlight onPress={() =>{console.log('press', song.id)}}>
                  <View key={song.id} style={{ flexDirection: "row" }}>
                    <View style={{ backgroundColor: song.color, padding: 5 }}>
                      red
                    </View>
                    <View>{song.title}</View>
                    <View>voted</View>
                  </View>
                </TouchableHighlight>*/}