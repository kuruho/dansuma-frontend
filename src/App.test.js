import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing', () => {
  const tree = renderer.create(
    <App
      items={[]}
      errors={{}}
      loading={true}
      filter={'Top'}
      overlayVisible={false}
      onOpenUrl={()=>{}}
      onLoadItems={()=>{}}
      onLoadMore={()=>{}}
      onToggleOverlay={()=>{}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing with Overlay visible', () => {
  const tree = renderer.create(
    <App
      items={[]}
      errors={{}}
      loading={false}
      filter={'Top'}
      overlayVisible={true}
      onOpenUrl={()=>{}}
      onLoadItems={()=>{}}
      onLoadMore={()=>{}}
      onToggleOverlay={()=>{}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});