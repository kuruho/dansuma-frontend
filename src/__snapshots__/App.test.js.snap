exports[`test renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <View />
  <View
    style={
      Object {
        "backgroundColor": "#ffb74d",
        "flex": 1,
      }
    }>
    <View />
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {
            "backgroundColor": "#ff9800",
          },
          Object {
            "height": 75,
            "paddingTop": 20,
          },
        ]
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
            Object {
              "height": 50,
            },
          ]
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
            }
          }>
          <Image
            source={1}
            style={
              Object {
                "width": 20,
              }
            } />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                  "paddingLeft": 4,
                },
              ]
            }>
            HAgnostic News
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "paddingLeft": 4,
                },
              ]
            }>
             
            ios
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "borderRadius": 20,
                  "paddingBottom": 8,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 8,
                },
                Object {
                  "backgroundColor": "#ff5722",
                },
              ],
            ]
          }
          testID={undefined}>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "paddingRight": 5,
                }
              }>
              Top
            </Text>
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="small" />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "borderRadius": 20,
                  "paddingBottom": 8,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 8,
                },
                null,
              ],
            ]
          }
          testID={undefined}>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "paddingRight": 5,
                }
              }>
              Latest
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "padding": 10,
              }
            }>
            ?
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <ScrollView
        className="scrollView"
        contentContainerStyle={
          Object {
            "paddingBottom": 2,
            "paddingTop": 2,
          }
        }
        scrollEventThrottle={1}
        style={Object {}}>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "rgb(223,223,223)",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "padding": 2,
                "paddingTop": 10,
              },
              Object {
                "justifyContent": "center",
                "paddingBottom": 10,
              },
            ]
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "borderRadius": 20,
                    "paddingBottom": 8,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 8,
                  },
                  Object {
                    "backgroundColor": "#eee",
                    "borderWidth": 0,
                  },
                ],
              ]
            }
            testID={undefined}>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-around",
                  },
                  Object {
                    "height": 20,
                  },
                ]
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#9e9e9e",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                } />
              <ActivityIndicator
                animating={true}
                color="#999999"
                hidesWhenStopped={true}
                size="small" />
            </View>
          </View>
        </View>
      </ScrollView>
    </View>
  </View>
</View>
`;

exports[`test renders without crashing with Overlay visible 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <View />
  <View
    style={
      Object {
        "backgroundColor": "#ffb74d",
        "flex": 1,
      }
    }>
    <View
      style={
        Object {
          "marginTop": 48,
          "position": "absolute",
          "zIndex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "rgba(52,52,52, 0.0)",
            "color": "white",
            "fontSize": 28,
            "marginLeft": 675,
            "zIndex": 3,
          }
        }>
        ⏏
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,80)",
            "height": 400,
            "marginTop": -17,
            "padding": 10,
            "width": 750,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
              }
            }>
            HAgnostic News
          </Text>
           is a simple Hacker News reader for the Web and a React Native app (Android / iOS).
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
              "marginTop": 20,
            }
          }>
          Made with ❤ by
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#fb8c00",
                    "fontWeight": "bold",
                    "padding": 10,
                  },
                  Object {},
                ]
              }>
               
              Luigi Maselli
            </Text>
          </Text>
          , source code:
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#fb8c00",
                    "fontWeight": "bold",
                    "padding": 10,
                  },
                  Object {},
                ]
              }>
               
              github.com/grigio/HAgnostic-News
            </Text>
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {
            "backgroundColor": "#ff9800",
          },
          Object {
            "height": 75,
            "paddingTop": 20,
          },
        ]
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
            Object {
              "height": 50,
            },
          ]
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
            }
          }>
          <Image
            source={1}
            style={
              Object {
                "width": 20,
              }
            } />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                  "paddingLeft": 4,
                },
              ]
            }>
            HAgnostic News
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "paddingLeft": 4,
                },
              ]
            }>
             
            ios
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "borderRadius": 20,
                  "paddingBottom": 8,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 8,
                },
                Object {
                  "backgroundColor": "#ff5722",
                },
              ],
            ]
          }
          testID={undefined}>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "paddingRight": 5,
                }
              }>
              Top
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "borderRadius": 20,
                  "paddingBottom": 8,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 8,
                },
                null,
              ],
            ]
          }
          testID={undefined}>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                  "paddingRight": 5,
                }
              }>
              Latest
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "padding": 10,
              }
            }>
            ?
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <ScrollView
        className="scrollView"
        contentContainerStyle={
          Object {
            "paddingBottom": 2,
            "paddingTop": 2,
          }
        }
        scrollEventThrottle={1}
        style={Object {}}>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "backgroundColor": "rgb(223,223,223)",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "padding": 2,
                "paddingTop": 10,
              },
              Object {
                "justifyContent": "center",
                "paddingBottom": 10,
              },
            ]
          }>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "borderRadius": 20,
                    "paddingBottom": 8,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 8,
                  },
                  Object {
                    "backgroundColor": "#eee",
                    "borderWidth": 0,
                  },
                ],
              ]
            }
            testID={undefined}>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-around",
                  },
                  Object {
                    "height": 20,
                  },
                ]
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#9e9e9e",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }>
                Load more
              </Text>
            </View>
          </View>
        </View>
      </ScrollView>
    </View>
  </View>
</View>
`;
